packages/suite/src/views/settings/device/FirmwareTypeChange.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { GITHUB_FW_COMMIT_URL } from '@trezor/urls';
+import { Translation, TrezorLink } from '@suite-components';
+import { ActionButton, ActionColumn, SectionItem, TextColumn } from '@suite-components/Settings';
+import { useDevice, useActions } from '@suite-hooks';
+import * as routerActions from '@suite-actions/routerActions';
+import { getFwType, getFwVersion, isBitcoinOnly } from '@suite-utils/device';
+import { Button, Tooltip } from '@trezor/components';
+import { useAnchor } from '@suite-hooks/useAnchor';
+import { SettingsAnchor } from '@suite-constants/anchors';
+
+const Version = styled.div`
+    span {
+        display: flex;
+        align-items: center;
+
+        > :last-child {
+            margin-left: 6px;
+        }
+    }
+`;
+
+const VersionTooltip = styled(Tooltip)`
+    display: inline-flex;
+`;
+
+interface FirmwareTypeProps {
+    isDeviceLocked: boolean;
+}
+
+export const FirmwareTypeChange = ({ isDeviceLocked }: FirmwareTypeProps) => {
+    const { device } = useDevice();
+    const { goto } = useActions({
+        goto: routerActions.goto,
+    });
+    const { anchorRef, shouldHighlight } = useAnchor(SettingsAnchor.FirmwareType);
+
+    if (!device?.features) {
+        return null;
+    }
+
+    const { revision } = device.features;
+    const currentFwVersion = getFwVersion(device);
+    const currentFwType = getFwType(device);
+    const actionButtonId = isBitcoinOnly(device)
+        ? 'TR_SWITCH_TO_UNIVERSAL'
+        : 'TR_SWITCH_TO_BITCOIN';
+
+    const handleAction = () => {
+        goto('firmware-type', {
+            params: { cancelable: true },
+        });
+    };
+
+    return (
+        <SectionItem
+            data-test="@settings/device/firmware-type"
+            ref={anchorRef}
+            shouldHighlight={shouldHighlight}
+        >
+            <TextColumn
+                title={<Translation id="TR_FIRMWARE_TYPE" />}
+                description={
+                    currentFwVersion ? (
+                        <Version>
+                            <Translation
+                                id="TR_YOUR_FIRMWARE_TYPE"
+                                values={{
+                                    version: (
+                                        <VersionTooltip content={revision} disabled={!revision}>
```

Comment:
Could be extracted into the body of the component with `useMemo`, so that the JXS output is clearer.

Reply:
I am with you, but this is tricky because of this:

```
if (!device?.features) {
        return null;
    }
```

I cannot place `useMemo` before this because I do not know the revision and I cannot place it after because a hook cannot follow a condition. Do you have any suggestion how to handle this better?

Reply:
There is also one more variable... it is doable but perhaps simpler as it is.